site/data/dynamo.ts

```diff
@@ -230,26 +236,152 @@ export const getDisruptionById = async (disruptionId: string): Promise<Disruptio
             },
         }),
     );
-
     if (!dynamoDisruption.Items) {
         return null;
     }
+    const isEdited = dynamoDisruption.Items.some((item) => (item.SK as string).includes("#EDIT"));
+    let info = dynamoDisruption.Items.find((item) => item.SK === `${disruptionId}#INFO`);
 
-    const info = dynamoDisruption.Items.find((item) => item.SK === `${disruptionId}#INFO`);
     const consequences = dynamoDisruption.Items.filter(
-        (item) => (item.SK as string).startsWith(`${disruptionId}#CONSEQUENCE`) ?? false,
+        (item) =>
+            ((item.SK as string).startsWith(`${disruptionId}#CONSEQUENCE`) && !(item.SK as string).includes("#EDIT")) ??
+            false,
     );
-
+    if (isEdited) {
+        info = dynamoDisruption.Items.find((item) => item.SK === `${disruptionId}#INFO#EDIT`) ?? info;
+        const editedConsequences = dynamoDisruption.Items.filter(
+            (item) =>
+                ((item.SK as string).startsWith(`${disruptionId}#CONSEQUENCE`) &&
+                    (item.SK as string).endsWith("#EDIT")) ??
+                false,
+        );
+        editedConsequences.forEach((editedConsequence) => {
+            const existingIndex = consequences.findIndex(
+                (c) => c.consequenceIndex === editedConsequence.consequenceIndex,
+            );
+            if (existingIndex > -1) {
+                consequences[existingIndex] = editedConsequence;
+            } else {
+                consequences.push(editedConsequence);
+            }
+        });
+    }
     const parsedDisruption = disruptionSchema.safeParse({
         ...info,
         consequences,
+        publishStatus: isEdited ? "EDITING" : (info?.publishStatus as string),
     });
-
     if (!parsedDisruption.success) {
         logger.warn(`Invalid disruption ${disruptionId} in Dynamo`);
-
         return null;
     }
-
     return parsedDisruption.data;
 };
+
+export const publishEditedConsequences = async (disruptionId: string) => {
+    const dynamoDisruption = await ddbDocClient.send(
+        new QueryCommand({
+            TableName: tableName,
+            KeyConditionExpression: "PK = :1 and begins_with(SK, :2)",
+            ExpressionAttributeValues: {
+                ":1": "1", // TODO: replace with user ID when we have auth,
+                ":2": `${disruptionId}`,
+            },
+        }),
+    );
+
+    if (dynamoDisruption.Items) {
+        const editedDisruption = dynamoDisruption.Items.filter(
```

Comment:
tried changing to find but the transact write command would need an iterable object which works with filter. If find is used then we will have to add separate if conditions so that transact write command does not contain empty objects